aea/cli/issue_certificates.py

```diff
@@ -0,0 +1,103 @@
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.configurations.constants import CONNECTION
+from aea.crypto.registries import crypto_registry
+from aea.exceptions import enforce
+from aea.helpers.base import CertRequest
+
+
+@click.command()
+@click.pass_context
+@check_aea_project
+def issue_certificates(click_context):
+    """Issue certificates for connections that require them."""
+    ctx = cast(Context, click_context.obj)
+
+    for connection_id in ctx.agent_config.connections:
+        _process_connection(ctx, connection_id)
+
+    click.echo("Done!")
+
+
+def _process_certificate(ctx: Context, cert_request: CertRequest):
+    """Process a single certificate request."""
+    click.echo(f"Issuing certificate '{cert_request.identifier}'...")
+    ledger_id = cert_request.ledger_id
+    output_path = cert_request.path
+    if cert_request.key_identifier is not None:
+        key_identifier = cert_request.key_identifier
+        connection_private_key_path = ctx.agent_config.connection_private_key_paths.read(
+            key_identifier
+        )
+        if connection_private_key_path is None:
+            raise ClickException(
+                f"Cannot find connection private key with id '{key_identifier}'"
+            )
+        connection_crypto = crypto_registry.make(
+            key_identifier, private_key_path=connection_private_key_path
+        )
+        public_key = connection_crypto.public_key
+    else:
+        public_key = cast(str, cert_request.public_key)
+        enforce(
+            public_key is not None,
+            "Internal error - one of key_identifier or public_key must be not None.",
+        )
+    public_key_bytes = public_key.encode("ascii")
+    identifier = cert_request.identifier.encode("ascii")
+    not_before = cert_request.not_before_string.encode("ascii")
+    not_after = cert_request.not_after_string.encode("ascii")
+    crytpo_private_key_path = ctx.agent_config.connection_private_key_paths.read(
+        ledger_id
+    )
+    if crytpo_private_key_path is None:
+        raise ClickException(f"Cannot find private key with id '{ledger_id}'")
+    crypto = crypto_registry.make(ledger_id, private_key_path=crytpo_private_key_path)
+    message = public_key_bytes + identifier + not_before + not_after
+    signature = crypto.sign_message(message).encode("ascii")
+    Path(output_path).write_bytes(signature)
+    click.echo(f"Dumped certificate in '{output_path}'.")
+
+
+def _process_connection(ctx: Context, connection_id: PublicId):
+    click.echo(f"Processing connection '{connection_id}'...")
+    path = get_package_path_unified(ctx, CONNECTION, connection_id)
+    connection_config = cast(ConnectionConfig, load_item_config(CONNECTION, Path(path)))
+    if (
+        connection_config.cert_requests is None
+        or len(connection_config.cert_requests) == 0
+    ):
+        click.echo("No certificates to process.")
+        return
```

Comment:
I don't think we should print this for each connection. Probably make it debug instead.